app/assets/javascripts/discourse/app/routes/build-category-route.js

```diff
@@ -56,6 +56,7 @@ export default (filterArg, params) => {
       if (
         (!params || params.no_subcategories === undefined) &&
         category.default_list_filter === "none" &&
+        modelParams !== undefined &&
```

Comment:
do we need explicit check on undefined?

what about just doing `&& modelParams &&`

Reply:
Explicit check is not a requirement. I just made the suggested change.